### ISSUE TITLE
[Face] Results getting worse after 50 epochs, model starts diverging

### ISSUE THREAD

**Issue by hamzatrq** [NVlabs/few-shot-vid2vid#26]
Logs:
[loss_log.txt](https://github.com/NVlabs/few-shot-vid2vid/files/4032621/loss_log.txt)

opt.txt
[opt.txt](https://github.com/NVlabs/few-shot-vid2vid/files/4032626/opt.txt)

Upto 50 epochs results were satisfactory but then the results started to get worse.

After 50 Epochs:
![00000](https://user-images.githubusercontent.com/13986516/71931935-b813ab00-31c0-11ea-8a74-38d5350a127f.jpg)
![00001](https://user-images.githubusercontent.com/13986516/71932173-2a848b00-31c1-11ea-8f94-df102c455dc0.jpg)

After 55 Epochs:
![00000](https://user-images.githubusercontent.com/13986516/71931874-974b5580-31c0-11ea-9efe-77d4ff72b523.jpg)
![00001](https://user-images.githubusercontent.com/13986516/71932160-2193b980-31c1-11ea-8204-0fff1d3e0256.jpg)

After 60 Epochs:
![00000](https://user-images.githubusercontent.com/13986516/71931887-a205ea80-31c0-11ea-92d3-deb33f110a3c.jpg)
![00001](https://user-images.githubusercontent.com/13986516/71932140-1771bb00-31c1-11ea-8707-2da0af7252e0.jpg)

After 65 Epochs:
![00000](https://user-images.githubusercontent.com/13986516/71931984-cf529880-31c0-11ea-890e-9b40ae2b0204.jpg)
![00001](https://user-images.githubusercontent.com/13986516/71932116-0de85300-31c1-11ea-942b-73f9900aec68.jpg)

After 75 Epoch:
![00001](https://user-images.githubusercontent.com/13986516/71932068-fc9f4680-31c0-11ea-8b53-7340bdcc031d.jpg)
![00000](https://user-images.githubusercontent.com/13986516/71932070-fc9f4680-31c0-11ea-8a9a-941d66d0ea10.jpg)

**Comment by hamzatrq** [comment]
Dataset Preprocessing:
I used Face Forensics dataset, downloaded the videos, renamed them, converted them to frames, used dlib to generate keypoints.

**Comment by nfrik** [comment]
How many images are there in your dataset? Did you have better results with the dataset provided by NVIDIA? In my case quality also degrades after epoch 50 and test images don't look that good either.

Epoch 50:
![iTerm2 EvoRtj epoch050_iter0489800_synthesized_image](https://user-images.githubusercontent.com/1614059/71934612-d488f680-3172-11ea-81f2-33f73ed15214.jpg)

Epoch 93:
![iTerm2 LtrNzk epoch093_iter0924500_synthesized_image](https://user-images.githubusercontent.com/1614059/71934641-e1a5e580-3172-11ea-89e4-873dfac8b129.jpg)

**Comment by hamzatrq** [comment]
@nfrik nvidia just provided example images, I do not know exactly the number of frames but I used 754 videos for training as mentioned in the paper.

**Comment by BaldrLector** [comment]
@hamzatrq 
Hi，from your loss log, G_GAN is increasing and DT is reducing since 50 epoch, it seems the model is collapsed.

**Comment by hamzatrq** [comment]
@BaldrLector do you think retraining the model will solve the problem? Also if yes. From 50 epochs or from start?

**Comment by BaldrLector** [comment]
@hamzatrq 
Hi, If epoch 50 works well then there no necessary to retrain the model,

In fact, I am also facing the same problem as you, the second training stage is so hard ( it is maybe the longer the sequence, the more varying rotation, translation, and zoom).

Some of my training results could be seen at #18, the second stage training indeed harm the performance. I guess there may be some trick to avoid model collapse,

**Comment by tcwang0509** [comment]
I will look into the issue once I get back. In the meantime, if epoch 50 works well, you can run inference using the model by adding `--n_frames_G 1` to force using single frame.

**Comment by lukemelas** [comment]
Authors, thank you for your wonderful paper and your responsiveness in replying to issues. 

I just wanted to note that I am also experiencing this issue -- face training diverges after epoch 50. I am using 1000 videos from the FaceForensics++ dataset and training on 8 GPUs with the provided hyperparameters.

**Comment by MengXinChengXuYuan** [comment]
I am also facing the same problem
After 50 epochs there shows black holes on  the syn image in my experiment

**Comment by ndyashas** [comment]
I have noticed that when I stop while in temporal training stage and then continue again, the time taken per iter reduces to what it was when I was in single frame training stage. Also the GPU's memory usage drops to what it was when in single frame training stage. I see similar trend in your log file @hamzatrq 
@tcwang0509

**Comment by akhilsantha7** [comment]
Hello, does any one solved the issue of diverging after 50 epochs..? Please do let me know. I am also facing the same issue.

Thanks,
Akhil.

**Comment by hamzatrq** [comment]
Hi Akhil, no still waiting for the author to reply.

**Comment by BaldrLector** [comment]
I am also waiting for the author to reply.

**Comment by hamzatrq** [comment]
@tcwang0509 can you please shed some light on the issue?
Thank you!

**Comment by BaldrLector** [comment]
I find a very related face reenactment work:
see   https://aliaksandrsiarohin.github.io/first-order-model-website/
same from NIPS, results are good, did anyone try it in the face dataset?

**Comment by tcwang0509** [comment]
There was a potential bug during refactoring that's fixed in the latest commit. Please try and see if it works now.

**Comment by tcwang0509** [comment]
This repo is now deprecated. Please refer to the new Imaginaire repo: https://github.com/NVlabs/imaginaire.
